### PR TITLE
[1.22.x] Automated cherry pick of #103958: Set idle and readheader timeouts

### PR DESCRIPTION
**Pull request by liggitt**
Cherry pick of #103958 on release-1.22.

#103958: Set idle and readheader timeouts

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

**Comment by liggitt** [comment]
/hold cancel

master PR has merged, but it's ok if this goes into 1.22.1

**Comment by ehashman** [comment]
Let's perhaps wait for this to bake a bit in 1.23 before backporting: https://kubernetes.slack.com/archives/C0EG7JC6T/p1627928455019100?thread_ts=1627927963.017700&cid=C0EG7JC6T

**Comment by liggitt** [comment]
> Let's perhaps wait for this to bake a bit in 1.23 before backporting: https://kubernetes.slack.com/archives/C0EG7JC6T/p1627928455019100?thread_ts=1627927963.017700&cid=C0EG7JC6T

I'm fine with that

**Comment by saschagrunert** [comment]
Clearing the milestone to avoid confusion.

/milestone clear

**Comment by cpanato** [comment]
@liggitt @saschagrunert this is good to get in? or waiting for something else? the other CPs lgtm, if this is not blocked i will cherry pick those

**Comment by liggitt** [comment]
@ehashman requested some soak time in master before backport. This is fixing an issue that has existed for a long time, so I don't object to a couple weeks of soak time, but I would like to make sure the backport lands in 1.19 before the branch shuts down. Is the upcoming 1.19 patch release the last one planned?

**Comment by cpanato** [comment]
> @ehashman requested some soak time in master before backport. This is fixing an issue that has existed for a long time, so I don't object to a couple weeks of soak time, but I would like to make sure the backport lands in 1.19 before the branch shuts down. Is the upcoming 1.19 patch release the last one planned?

yes, for this week, I will take a look on the CP for the other branches, thanks @liggitt

**Comment by cpanato** [comment]
from parent PR:
/priority important-soon

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by cpanato** [comment]
@liggitt i think you are the only one to lift the hold

**Comment by cpanato** [comment]
oh the cancel worked 😕

**Pull request merged**
